Assets/Scripts/Game/Enemy.cs

```diff
@@ -20,7 +20,9 @@ protected override void Awake()
 
     void Start()
     {
-        OnHealthChanged.AddListener(FlashRed);
+        OnDamageTaken.AddListener(FlashRed);
+        OnDamageTaken.AddListener((health) => GameManager.LevelingSystem.AddXP(20));
+        OnDeath.AddListener(() => GameManager.LevelingSystem.AddXP(100));
     }
```

Comment:
Why do you get XP for taking damage and dying?

Reply:
This is the `Enemy` class, so this would be the enemy taking damage and dying that would trigger the XP gains, not the player taking damage/dying. There's some test enemies in the scene if you want to try it out.

Reply:
Oh true. That's my bad, I totally missed the name of the class.